src/ServiceControl/MessageFailures/Api/UnArchiveMessages.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace ServiceControl.MessageFailures.Api
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using InternalMessages;
+    using Nancy;
+    using Nancy.ModelBinding;
+    using NServiceBus;
+    using ServiceBus.Management.Infrastructure.Nancy.Modules;
+
+    public class UnArchiveMessages : BaseModule
+    {
+        public IBus Bus { get; set; }
+
+        public UnArchiveMessages()
+        {
+            Patch["/errors/unarchive"] = _ =>
+            {
+                var ids = this.Bind<List<string>>();
+
+                if (ids.Any(string.IsNullOrEmpty))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                var request = new InternalMessages.UnArchiveMessages { FailedMessageIds = ids };
+
+                Bus.SendLocal(request); 
+
+                return HttpStatusCode.Accepted;
+            };
+
+            Patch["/errors/{from}...{to}/unarchive"] = parameters =>
+            {
+                DateTime from, to;
+
+                try
+                {
+                    from = DateTime.Parse(parameters.from, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);
```

Comment:
Do we accept requests where from >= to? And what happens when we execute the query that way around?